code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceActivate.java

```diff
@@ -211,10 +213,16 @@ public Object call() throws Exception {
         return ntwkIds;
     }
 
-    protected Long getImage(String imageUuid) {
+    protected Long getImage(String imageUuid, Object registryCredentialId) {
```

Comment:
I can change getImage signature to accept launchConfig - it would probably be more clear. Then we retrieve imageId and registryCredential inside the method. Whatever you prefer.